src/ngMock/angular-mocks.js

```diff
@@ -556,9 +556,13 @@ angular.mock.$IntervalProvider = function() {
      */
     $interval.flush = function(millis) {
       now += millis;
-      while (repeatFns.length && repeatFns[0].nextTime <= now) {
+      var firstTask = repeatFns[0];
+      while (repeatFns.length && (repeatFns[0].nextTime <= now || !repeatFns[0].nextTime)) {
         var task = repeatFns[0];
         task.fn();
+        if (!task.delay) {
+          task.delay = task.nextTime = 1;
```

Comment:
If `delay` is zero then `nextTime` will also be zero. Setting to 1 means the next round it'll be 2 since in fact will be the round number 2. Thanks to first set to 1 and then increasing it means that passing a value of 1 to `ms` will run only once and not twice the function. I hope this makes sense :-)